dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.utils
+
+import com.datadog.android.utils.forge.Configurator
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import kotlin.system.measureNanoTime
+
+private const val ITEMS_TO_JOINT = 10_000
+private const val REPETITION_COUNT = 10_000
+
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+@ForgeConfiguration(Configurator::class)
+internal class PerformanceTest {
+
+    private fun List<Long>.mean() = sum().toDouble() / size
+
+    @Test
+    fun `M be faster than joinToString W buildString`(forge: Forge) {
+        val itemsForJoin = forge.aList(ITEMS_TO_JOINT) { forge.aString() }
```

Comment:
> [!NOTE]
> You're running all the iterations on the same dataset. On the one hand, it gives a more relevant mean, but I'm curious whether we could get more insights if we moved the `itemsForJoin` inside the loop and have a different data set each time. 
> Also we're looking only at the mean, what about other insights (p90, stdev, etc…), can we learn info about edge cases from those?

Reply:
Moving `itemsForJoin` creation inside the loop will significantly increase test execution time, so I left it outside for now. 

Regarding the p95 - it's a good point. After some research I've noticed that the STD value is relatively big, so mean is not the best metric for this measurement. I've changed  metric to p95 as it seems good compromise between accuracy and sustainability to flakiness. Still test is highly depends on the machine state, so if we will notice that it flaky will reduce metric to p90

Reply:
Upd: reduced to p90 for more stability